e2e/run_test.go

```diff
@@ -28,12 +39,132 @@ func kubectlWithInput(input []byte, args ...string) ([]byte, []byte, error) {
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal MySQLCluster. stdout: %s, err: %v", stdout, err)
+	}
+	return &mysqlCluster, nil
+}
+
+func getRootPassword(mysqlCluster *mocov1alpha1.MySQLCluster) (*corev1.Secret, error) {
+	stdout, stderr, err := kubectl("get", "-n", "e2e-test", "secret", "root-password-"+moco.UniqueName(mysqlCluster), "-o", "json")
+	if err != nil {
+		return nil, fmt.Errorf("failed to get Secret. stdout: %s, stderr: %s, err: %v", stdout, stderr, err)
+	}
+
+	var secret corev1.Secret
+	err = json.Unmarshal(stdout, &secret)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal Secret stdout: %s, err: %v", stdout, err)
+	}
+	return &secret, nil
+}
+
+type mysqlConnector struct {
+	cluster             *mocov1alpha1.MySQLCluster
+	portForwardCommands []*exec.Cmd
+	accessor            *accessor.MySQLAccessor
+	basePort            int
+}
+
+func newMySQLConnector(cluster *mocov1alpha1.MySQLCluster) *mysqlConnector {
+	acc := accessor.NewMySQLAccessor(&accessor.MySQLAccessorConfig{
+		ConnMaxLifeTime:   30 * time.Minute,
+		ConnectionTimeout: 3 * time.Second,
+		ReadTimeout:       30 * time.Second,
+	})
+	return &mysqlConnector{
+		cluster:  cluster,
+		accessor: acc,
+		basePort: 13306,
+	}
+}
+
+func (c *mysqlConnector) startPortForward() error {
+	for i := 0; i < int(c.cluster.Spec.Replicas); i++ {
+		podName := fmt.Sprintf("%s-%d", moco.UniqueName(c.cluster), i)
+		port := c.basePort + i
+		command := exec.Command("./bin/kubectl", "-n", "e2e-test", "port-forward", "pod/"+podName, fmt.Sprintf("%d:%d", port, moco.MySQLPort))
+		command.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+		err := command.Start()
+		if err != nil {
+			c.stopPortForward()
+			return err
+		}
+		c.portForwardCommands = append(c.portForwardCommands, command)
+	}
+	return nil
+}
+
+func (c *mysqlConnector) stopPortForward() {
+	for _, command := range c.portForwardCommands {
+		_ = command.Process.Kill()
+	}
+	c.portForwardCommands = []*exec.Cmd{}
+}
+
+func (c *mysqlConnector) connectToPrimary() (*sqlx.DB, error) {
+	if c.cluster.Status.CurrentPrimaryIndex == nil {
+		return nil, errors.New("CurrentPrimaryIndex is nil")
+	}
+	return c.connect(*c.cluster.Status.CurrentPrimaryIndex)
+}
+
+func (c *mysqlConnector) connect(index int) (*sqlx.DB, error) {
+	port := c.basePort + index
+	addr := fmt.Sprintf("127.0.0.1:%d", port)
+	secret, err := getRootPassword(c.cluster)
+	if err != nil {
+		return nil, err
+	}
+	password := string(secret.Data[moco.OperatorPasswordEnvName])
+	return c.accessor.Get(addr, moco.OperatorAdminUser, password)
+}
+
+func findCondition(conditions []mocov1alpha1.MySQLClusterCondition, conditionType mocov1alpha1.MySQLClusterConditionType) *mocov1alpha1.MySQLClusterCondition {
+	for i, c := range conditions {
+		if c.Type == conditionType {
+			return &conditions[i]
 		}
 	}
-	return false
+	return nil
+}
+
+func replica0(cluster *mocov1alpha1.MySQLCluster) (int, error) {
```

Comment:
what do you mean by this func name `replica0`? It seems ambiguous.
I recommend a name such as `minIndexReplica`.